When `add_install_script` is used with `./` (curr.dir), the script may fail to run

**Describe the bug**

When using `meson.add_install_script('./test.sh')`, depending on the directory tree, it may happen that the script could not be run upon running later `ninja install` or `meson install`.

**To Reproduce**

```
 λ cat test.sh
#!/bin/bash
echo hello
 λ ./test.sh               # NOTE: the script itself runs fine
hello
 λ cat meson.build
project(
  'test',
  ['c'],
)
subdir('dir1/dir2/')
 λ cat dir1/dir2/meson.build
meson.add_install_script('./test.sh')
 λ meson build
The Meson build system
Version: 0.59.1
Source dir: /tmp/foo
Build dir: /tmp/foo/build
Build type: native build
Project name: test
Project version: undefined
C compiler for the host machine: ccache cc (gcc 11.1.0 "cc (GCC) 11.1.0")
C linker for the host machine: cc ld.bfd 2.36.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program ./test.sh found: YES (./test.sh)
Build targets in project: 0

Found ninja-1.10.2 at /usr/bin/ninja
 λ DESTDIR=/tmp/ ninja -C build install
ninja: Entering directory `build'
[0/1] Installing files.
Running custom install script './test.sh'
FAILED: install script './test.sh' could not be run, stopped
FAILED: meson-install
/usr/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
```

**Expected behavior**

The last command would succeed

**Known Workarounds**

With the directory tree here, if script is used as `meson.add_install_script('../../test.sh')`, then there will be no errors upon installation.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build ? Native
* OS: Archlinux
* Python version: 3.9.6
* `meson --version`: 0.59.1
* `ninja --version`: 1.10.2

--- a/mesonbuild/programs.py
+++ b/mesonbuild/programs.py
@@ -295,6 +295,9 @@ class ExternalProgram(mesonlib.HoldableObject):
         commands = self._search_dir(name, search_dir)
         if commands:
             return commands
+        # If there is a directory component, do not look in PATH
+        if os.path.dirname(name) and not os.path.isabs(name):
+            return [None]
         # Do a standard search in PATH
         path = os.environ.get('PATH', None)
         if mesonlib.is_windows() and path:

